src/components/vm/nics/nicEdit.tsx

```diff
@@ -168,24 +168,34 @@ export class EditNICModal extends React.Component<EditNICModalProps, EditNICModa
             return;
         }
 
-        domainChangeInterfaceSettings({
-            vmName: vm.name,
-            connectionName: vm.connectionName,
-            persistent: vm.persistent,
-            macAddress: network.mac || "",
-            newMacAddress: this.state.networkMac,
-            networkModel: this.state.networkModel,
-            networkType: this.state.networkType,
-            networkSource: this.state.networkSource,
-            networkSourceMode: this.state.networkSourceMode,
-        })
-                .then(() => {
-                    domainGet({ connectionName: vm.connectionName, id: vm.id });
-                    Dialogs.close();
-                })
-                .catch((exc) => {
-                    this.dialogErrorSet(_("Network interface settings could not be saved"), exc.message);
-                });
+        try {
+            let source;
+            if (this.state.networkType == "direct") {
+                source = {
+                    "": this.state.networkSource,
+                    mode: this.state.networkSourceMode,
+                };
+            } else {
+                source = this.state.networkSource;
+            }
+            await virtXmlEdit(
+                vm,
+                "network",
+                {
+                    mac: network.mac
+                },
+                {
+                    mac: this.state.networkMac,
+                    model: this.state.networkModel,
+                    type: this.state.networkType,
+                    source,
+                }
+            );
+            domainGet({ connectionName: vm.connectionName, id: vm.id });
+            Dialogs.close();
+        } catch (exc) {
+            this.dialogErrorSet(_("Network interface settings could not be saved"), String(exc));
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2334-68be91e9-20250929-092324-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)